src/Momento.Sdk/Internal/DataGrpcManager.cs

```diff
@@ -18,12 +22,24 @@ public class DataGrpcManager : IDisposable
     // https://learn.microsoft.com/en-us/dotnet/api/system.environment.version?view=netstandard-2.0#remarks
     private readonly string runtimeVersion = "dotnet:" + System.Environment.Version;
 
-    internal DataGrpcManager(string authToken, string host)
+    internal DataGrpcManager(IList<IMiddleware> middlewares, string authToken, string host)
     {
         var url = $"https://{host}";
         this.channel = GrpcChannel.ForAddress(url, new GrpcChannelOptions() { Credentials = ChannelCredentials.SecureSsl });
         List<Header> headers = new List<Header> { new Header(name: Header.AuthorizationKey, value: authToken), new Header(name: Header.AgentKey, value: version), new Header(name: Header.RuntimeVersionKey, value: runtimeVersion) };
-        CallInvoker invoker = this.channel.Intercept(new HeaderInterceptor(headers));
+
+
+        CallInvoker invokerWithMiddlewares = middlewares.Aggregate(
+            this.channel.CreateCallInvoker(),
+            (invoker, middleware) =>
+            {
+                Console.WriteLine($"Adding an interceptor: {middleware}");
+                return invoker.Intercept(new MiddlewareInterceptor(middleware));
+            }
+        );
+
```

Comment:
Will need to make clear in the configuration documentation that the order the middlewares execute will be last-to-first

Reply:
good callout.